v4/newrelic/transaction.go

```diff
@@ -257,9 +306,15 @@ func (txn *Transaction) BrowserTimingHeader() *BrowserTimingHeader {
 // Note that any segments that end after the transaction ends will not
 // be reported.
 func (txn *Transaction) NewGoroutine() *Transaction {
+	txn.thread.Lock()
+	defer txn.thread.Unlock()
+
+	txn.thread.isMultiSpan = true
```

Comment:
Oh nice, I hadn't thought of this change but it makes total sense. This is great.